app/src/Controller/ArticlesController.php

```diff
@@ -31,63 +32,64 @@ public function view($slug = null)
         ->findBySlug($slug)
         ->contain('Tags')
         ->firstOrFail();
+
+        $this->Authorization->skipAuthorization();
+
         $this->set(compact('article'));
     }
 
     public function add()
     {
         $article = $this->Articles->newEmptyEntity();
+        $this->Authorization->authorize($article);
+
         if ($this->request->is('post')) {
             $article = $this->Articles->patchEntity($article, $this->request->getData());
 
-            // Hardcoding the user_id is temporary, and will be removed later
-            // when we build authentication out.
-            $article->user_id = 1;
+            // Changed: Set the user_id from the current user.
+            $article->user_id = $this->request->getAttribute('identity')->getIdentifier();
```

Comment:
Alternative way to get the logged in user is through Authentication component. Refer https://book.cakephp.org/authentication/2/en/authentication-component.html#accessing-the-logged-in-user

> You can get the authenticated user identity data using the authentication component:

Eg:

```
            $article->user_id = $this->Authentication->getIdentity()?->getIdentifier();
```

Reply:
removed